src/parsers/xyz.js

```diff
@@ -140,10 +140,26 @@ function fromMaterial(materialOrConfig, fractional = false) {
     return fromBasis(basis, "%11.6f");
 }
 
+/**
+ * Function returns the string on the second line of XYZ file content. Generally the second line contains text
+ * containing the name of the structure.
+ * @param {String} name
+ * @param {String} fileContent
+ * @returns {String}
+ */
+export function materialNameFromFileContents(name, fileContent) {
+    const materialNameFromFile = fileContent.split(/\r?\n/)[1];
+    if (name === materialNameFromFile || !materialNameFromFile) {
+        return name;
+    }
+    return materialNameFromFile;
```

Comment:
1. Add a general utility - `getLineFromContent(lineNumber)`
2. `materialNameFromFileContents` -> `getNameFromContents`
3. `getNameFromContents(fileContent) => {return getLineFromContent(1)}` for POSCAR, and `0` for XYZ
The default name and everything on lines 152-154 should not be dealt with here. We could add a function named: `getNameFromFIleContent(type, fileContent, failoverName = "material")` and use as a proxy.